src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActorContext.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Context object passed to application code executing actor operations.
+    /// </summary>
+    internal class DurableActorContext : DurableCommonContext, IDurableActorContext
+    {
+        private readonly ActorId self;
+
+        public DurableActorContext(DurableTaskExtension config, ActorId actor)
+         : base(config, actor.ActorClass)
+        {
+            this.self = actor;
+        }
+
+        internal SchedulerState State { get; set; }
+
+        internal OperationMessage CurrentOperation { get; set; }
+
+        internal ResponseMessage CurrentOperationResponse { get; set; }
+
+        internal bool Fresh { get; set; }
+
+        internal bool DestructOnExit { get; set; }
+
+        string IDurableActorContext.ActorClass => this.self.ActorClass;
+
+        string IDurableActorContext.Key => this.self.ActorKey;
+
+        ActorId IDurableActorContext.Self => this.self;
+
+        string IDurableActorContext.OperationName
+        {
+            get
+            {
+                this.ThrowIfInvalidAccess();
+                return this.CurrentOperation.Operation;
+            }
+        }
+
+        ILogger IDurableActorContext.Logger => throw new NotImplementedException();
+
+        bool IDurableActorContext.NewlyConstructed
+        {
+            get
+            {
+                this.ThrowIfInvalidAccess();
+                return this.Fresh;
+            }
+        }
+
+        void IDurableActorContext.DestructOnExit()
+        {
+            this.ThrowIfInvalidAccess();
+            this.DestructOnExit = true;
+        }
+
+        TArgument IDurableActorContext.GetOperationContent<TArgument>()
+        {
+            this.ThrowIfInvalidAccess();
+            return this.CurrentOperation.GetContent<TArgument>();
+        }
+
+        object IDurableActorContext.GetOperationContent(Type argumentType)
+        {
+            this.ThrowIfInvalidAccess();
+            return this.CurrentOperation.GetContent(argumentType);
+        }
+
+        IStateView<TState> IDurableActorContext.GetStateAs<TState>(Formatting formatting, JsonSerializerSettings settings)
+        {
+            this.ThrowIfInvalidAccess();
+            if (this.State.CurrentStateView != null)
+            {
+                if (this.State.CurrentStateView is IStateView<TState> view)
```

Comment:
Sure. The point is that applications can choose to use multiple different .NET types to create different views of the same JSON object. This can happen when applications are migrating between versions, for example. Also, applications can create views on part of the state, rather than the whole state, by customizing the JSON serialization settings so they work on a subset of the fields of the object.